File: src/hooks/maestro.hooks.ts
import { Before } from '@cucumber/cucumber';
import { ConductorWorld } from '../world/ConductorWorld';

Before({ tags: '@mobile or @cross-platform' }, async function (this: ConductorWorld) {
  const deviceId = this.config.mobile.deviceId ?? 'default';
  this.logger.info({ deviceId }, 'Targeting Maestro device');
});

File: java/conductor-core/src/main/java/com/nouhouari/conductor/hooks/MaestroHooks.java
package com.nouhouari.conductor.hooks;

import com.nouhouari.conductor.world.ConductorWorld;
import io.cucumber.java.Before;

/** Java port of src/hooks/maestro.hooks.ts. No @After — Maestro has no persistent process to tear down. */
public class MaestroHooks {

    private final ConductorWorld world;

    public MaestroHooks(ConductorWorld world) {
        this.world = world;
    }

    @Before("@mobile or @cross-platform")
    public void logTargetDevice() {
        String deviceId = world.config.mobile().deviceId() != null ? world.config.mobile().deviceId() : "default";
        world.logger.info("Targeting Maestro device: {}", deviceId);
    }
}
